Added the data portal navigation bar to explorer

atarashansky opened the pull request:
Closes issue #383 

#### Reviewers
**Functional:** 
@tihuan @seve 
**Readability:** 
@hthomas-czi @signechambers1 
---

## Changes
- added the unified navigation bar to the top of explorer
- moved the information hamburger menu to the unified nav bar
- removed the cellxgene logo

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/16548075/183938306-577f9d6e-b545-4915-82b5-0281c8f0d3b4.png">

atarashansky commented:
Fonts are a little different in the nav bar explorer but i'm not sure if that's due to local vs prod differences

https://user-images.githubusercontent.com/16548075/183941198-f02894f6-804a-47c7-be39-0b9b90ce3021.mov

atarashansky commented on client/src/components/NavBar/style.ts:
```diff
@@ -0,0 +1,86 @@
+    background: none;
+    border-radius: 0;
+    color: ${GRAY.D};
+    font-size: 13px;
```
Adjusting font-size doesn't seem to do anything.

tihuan replied:
You mean changing the font size doesn't make it match the data portal navbar?

Can you use chrome devtool to inspect the css diff between explorer and DP to see if there's any difference? Another possibility is that Explorer is not downloading the font file needed :thinking:

atarashansky replied:
I meant that changing the font size here isn't changing it in the UI - I tried decreasing it to make it match the data portal navbar.

tihuan replied:
Ah got it!

I have a fix [PR here](https://github.com/chanzuckerberg/single-cell-explorer/pull/386) PTAL thank you!

atarashansky replied:
you da man!

tihuan commented on client/package.json:
```diff
@@ -81,6 +81,7 @@
     "regenerator-runtime": "^0.13.7",
     "regl": "^2.1.0",
     "sha1": "^1.1.1",
+    "styled-components": "^5.3.5",
```
Explorer already has Emotion, so we don't need Styled Components 👌

tihuan commented on client/src/analytics/events.ts:
```diff
@@ -48,4 +48,5 @@ export enum EVENTS {
   EXPLORER_ADD_GENE_AND_SELECT_HISTOGRAM = "EXPLORER_ADD_GENE_AND_SELECT_HISTOGRAM",
   EXPLORER_SELECT_HISTOGRAM = "EXPLORER_SELECT_HISTOGRAM",
   EXPLORER_GENE_INFO_BUTTON_CLICKED = "EXPLORER_GENE_INFO_BUTTON_CLICKED",
+  WMG_CLICK_NAV = "WMG_CLICK_NAV",
```
We might need to prefix this event with `EXPLORER_`, so Amanda can tell whether the click is from Explorer or Data Portal. Or if she doesn't care then never mind!

atarashansky replied:
I asked - prefix not necessary as plausible tracks source URL

tihuan commented on client/src/components/leftSidebar/index.tsx:
```diff
@@ -7,7 +7,6 @@ import Categorical from "../categorical";
 import * as globals from "../../globals";
 import DynamicScatterplot from "../scatterplot/scatterplot";
 import GeneInfo from "../geneExpression/geneInfo/geneInfo";
-import TopLeftLogoAndTitle from "./topLeftLogoAndTitle";
```
Do we need to remove this from `client/src/components/leftSidebar/leftSidebarSkeleton.tsx` as well?

https://github.com/chanzuckerberg/single-cell-explorer/search?q=TopLeftLogoAndTitle&type=

tihuan reviewed (commented):
Looks really good! Just some tiny comments :rocket: :sparkles: Thanks so much for adding this, Alec :pray: !

seve commented:
Two design/product questions for @signechambers1 and @hthomas-czi:

1. Now that we have documentation in the header, can we drop the “?” button that's in the bottom right corner of the graph?
   1. Regardless, we should have the documentation link in the header be explorer specific, like how the “?” currently is
2. The header takes up a decent amount of vertical space, but isn't needed a vast majority of the time. Can we introduce hiding?

seve commented:
FYI: You're probably going to see test failures because the size of the graph is changing due to changes

hthomas-czi commented:
@seve 
1. Yes, we should definitely drop the "?" button in the bottom right.
2. That's a good idea, but I'll need some time to explore some solutions.

@atarashansky 
1. The hamburger menu should be consolidated into the Help & Documentation button which will need to be a dropdown.
2. We should move the embedding selector from the bottom left to the top left. We should also update the icon (attached).
3. We should move the dataset selector from the top left to the bottom left. Also, remove the "Home" portion of the dataset selector and update the styling to match the rest of the interface.

This ([Figma](https://www.figma.com/file/qpE40pnzlVal1U1Oib9EKg/Unified-Navigation?node-id=194%3A52616)) file should show everything I mentioned. Let me know if you have any questions!

![scatterplot](https://user-images.githubusercontent.com/72052788/183983565-2155757b-673c-43e7-bd68-93febcbffae4.png)
<img width="16" alt="scatterplot@2x" src="https://user-images.githubusercontent.com/72052788/183983567-9347c7a2-2a02-4ca5-8009-e3b4806a3940.png">

hthomas-czi commented on client/src/components/NavBar/theme.ts:
```diff
@@ -0,0 +1,11 @@
+export const PT_GRID_SIZE_PX = 8;
+
+export const PT_TEXT_COLOR = "#10161A";
```
Please use the Figma file as the reference for styling rather than the existing navigation bar. We should update the existing navigation bar to match this implementation, if necessary.

All text styles and colors should reference the Single Cell theme file at `frontend/src/common/theme.ts`. You can see the corresponding variables in Figma (e.g., `Gray/300` or `fontBodyS/600`)

atarashansky commented on client/src/components/embedding/index.tsx:
```diff
@@ -16,10 +16,124 @@ import actions from "../../actions";
 // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
 type EmbeddingState = any;
 
+const heatmapIcon = (
```
@tihuan I was having trouble importing an svg file, so I temporarily added the svg component in-file. Declaring it in a `.d.ts` file and adding it to `tsconfig.json` did not help... any tips?

tihuan replied:
Finally fixed everything I think lol

https://github.com/chanzuckerberg/single-cell-explorer/pull/391 PTAL thank you!

atarashansky commented:
- I updated the position of the embedding choice menu and added new icon.
 - I removed the home button from the dataset selector.
   - I'm not sure how to test this locally. I will ask @MillenniumFalconMechanic how she did it.
 - I made the help and documentation a dropdown menu via `Popover2`

seve commented on client/configuration/webpack/webpack.config.prod.js:
```diff
@@ -56,11 +56,8 @@ const prodConfig = {
       {
         test: /\.(jpg|png|gif|eot|svg|ttf|woff|woff2|otf)$/i,
         loader: "file-loader",
-        include: [nodeModules, fonts],
         options: {
           name: "static/assets/[name]-[contenthash].[ext]",
-          // (thuang): This is needed to make sure @font url path is '../static/assets/'
-          publicPath: "..",
```
Did you test this change? @sienacizdziel and I made this change after trying to get fonts to work for a few hours

tihuan replied:
I tested it by running `make build-for-server-dev && make start-server` and visit `http://localhost:5005/d/pbmc3k.cxg/`, and the static assets path match what dev/staging env have!

tihuan replied:
More details here: https://github.com/chanzuckerberg/single-cell-explorer/pull/391

tihuan replied:
context for keeping Roboto Condensed in the meantime:

Harley: "Inter takes up more horizontal space, so gene names were getting truncated with fewer characters. I still think we should switch to Inter, but I’d like to make a couple improvements to the gene UX to make more space for the gene name (see screenshot). Until we make those changes, Arathi wanted to revert back to Roboto Condensed."

<img width="1440" alt="Explorer" src="https://user-images.githubusercontent.com/6309723/189437682-77e77632-7fb3-4152-b684-1014cbbef3f0.png">

seve commented on client/package.json:
```diff
@@ -126,6 +126,7 @@
     "@types/react-helmet": "^6.1.2",
     "@types/react-redux": "^7.1.18",
     "@types/sha1": "^1.1.3",
+    "@types/styled-components": "^5.1.25",
```
```suggestion
```

seve reviewed (changes requested):
We'll need to revert all the font changes, since we're reverting the move to Inter on the Explorer

tihuan reviewed (approved):
Taking over this PR, so LGTMing myself lol

tihuan commented:
@seve @hthomas-czi PTAL again 🙏 Thank you!

atarashansky commented:
Thanks for carrying it to the finish line, @tihuan !!!

tihuan commented:
Thanks for doing 99% of the work for the feature too, Alec 😆 🙏 🚀

signechambers1 commented:
Thank you both! Anyway @hthomas-czi and I can review before it hits staging? The screenshot looks good but want to check the rest of the items in the ticket.

tihuan commented:
@signechambers1 ooh yes that's a great idea! @atarashansky do you remember if we could spin up rdev for Explorer? If so, mind doing that for me please 🙏

If rdev is not available, I guess we could merge this to deploy to dev for testing?

Thanks all!

signechambers1 commented:
I won't have time to review before I leave for class today so please make sure @hthomas-czi signs off while im out on vacation! thank you all!

tihuan commented:
Building an rdev now 🚀 Will share link here once complete!

tihuan commented:
Got it! Sounds good 👍 Enjoy your vacation, Signe!! 🍹 🎉 🛫

tihuan commented:
Somehow my rdev is not working. So maybe let's just merge this and check dev!

https://thuang-nav-explorer.rdev.single-cell.czi.technology/

tihuan commented:
Actually might need to upgrade Dockerfile to download node16!

https://github.com/chanzuckerberg/single-cell-explorer/pull/448

tihuan commented:
Example here:

https://cellxgene.dev.single-cell.czi.technology/e/e6722845-9377-4b34-80a2-7921921e1396.cxg/

CC: @hthomas-czi 

Thank you!
